src/bgw/job_stat.c

```diff
@@ -203,9 +203,9 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 	float8 multiplier = (consecutive_failures > MAX_FAILURES_MULTIPLIER ? MAX_FAILURES_MULTIPLIER :
 																		  consecutive_failures);
 	MemoryContext oldctx;
-	Assert(consecutive_failures > 0);
-	int64 max_slots =
-		(1 << consecutive_failures) - 1; /* 2^(consecutive_failures) - 1, at most 2^20 */
+	Assert(consecutive_failures > 0 && consecutive_failures < 63);
+	/* 2^(consecutive_failures) - 1, at most 2^20 */
+	int64 max_slots = (INT64CONST(1) << consecutive_failures) - INT64CONST(1);
```

Comment:
This line was introduced by me, and I realize now it should have been `1 << multiplier` instead (which is at most 20) and so the assertion should also be multiplier < 63.

Reply:
If you suggest the fix here, I'll incorporate that as a separate commit in the PR with your name on it and we can push them together.